Gems/AtomLyIntegration/CommonFeatures/Code/Source/Mesh/MeshComponentController.cpp

```diff
@@ -699,6 +701,81 @@ namespace AZ
+        {
+            // Attempt to copy the triangle list geometry data out of the model asset into the visible geometry structure
+            const auto& modelAsset = GetModelAsset();
+            if (!modelAsset.IsReady() || modelAsset->GetLodAssets().empty())
+            {
+                AZ_Warning("MeshComponentController", false, "Unable to get geometry because mesh asset is not ready or empty.");
+                return;
+            }
+
+            // This will only extract data from the first LOD. It might be necessary to make the LOD selectable.
+            const auto& lodAsset = modelAsset->GetLodAssets().front();
+            if (!lodAsset.IsReady())
+            {
+                AZ_Warning("MeshComponentController", false, "Unable to get geometry because selected LOD asset is not ready.");
+                return;
+            }
+
+            const AZ::Name positionName{ "POSITION" };
+            for (const auto& mesh : lodAsset->GetMeshes())
+            {
+                // Get the index buffer data, confirming that the asset is valid and indices are 32 bit integers. Other formats are
+                // currently not supported.
+                const AZ::RPI::BufferAssetView& indexBufferView = mesh.GetIndexBufferAssetView();
+                const AZ::RHI::BufferViewDescriptor& indexBufferViewDesc = indexBufferView.GetBufferViewDescriptor();
+                const AZ::Data::Asset<AZ::RPI::BufferAsset> indexBufferAsset = indexBufferView.GetBufferAsset();
+                if (!indexBufferAsset.IsReady() || indexBufferViewDesc.m_elementSize != sizeof(uint32_t))
+                {
+                    AZ_Warning("MeshComponentController", false, "Unable to get geometry for mesh because index buffer asset is not ready or is an incompatible format.");
+                    continue;
+                }
+
+                // Get the position buffer data, if it exists with the expected name.
+                const AZ::RPI::BufferAssetView* positionBufferView = mesh.GetSemanticBufferAssetView(positionName);
+                if (!positionBufferView)
+                {
+                    AZ_Warning("MeshComponentController", false, "Unable to get geometry for mesh because position buffer data was not found.");
+                    continue;
+                }
+
+                // Confirm that the position buffer is valid and contains 3 32 bit floats for each position. Other formats are currently not
+                // supported.
+                constexpr uint32_t ElementsPerVertex = 3;
+                const AZ::RHI::BufferViewDescriptor& positionBufferViewDesc = positionBufferView->GetBufferViewDescriptor();
+                const AZ::Data::Asset<AZ::RPI::BufferAsset> positionBufferAsset = positionBufferView->GetBufferAsset();
+                if (!positionBufferAsset.IsReady() || positionBufferViewDesc.m_elementSize != sizeof(float) * ElementsPerVertex)
```

Comment:
you could also check positionBufferViewDesc.m_elementFormat == RHI::Format::R32G32B32_FLOAT